### PR TITLE
Analysis::color()

### PR DESCRIPTION
**Pull request by romanzes**
I'm sure the practices I used here are not the best, so I hope you'll be able to help me make it all better.

### CODE REVIEW

File: src/auto/color.rs [hunk 1]
```diff
@@ -31,6 +31,18 @@ impl Color {
     fn to_string(&self) -> GString {
         unsafe { from_glib_full(pango_sys::pango_color_to_string(self.to_glib_none().0)) }
     }
+
+    pub fn red(&self) -> u16 {
```
Comment (romanzes) [inline src/auto/color.rs:35]: This should probably go into a new file outside of auto crate, right?

Comment (velyan) [reply src/auto/color.rs:35]: Yeah, otherwise it will get lost if this file gets regenerated

**Comment by velyan** [comment]
Added some changes based on yours that I think should be easier to merge - https://github.com/velyan/pango/compare/rp-color...velyan:rp-vel-color 

The idea is to be used like this:

```
                let attrs = analysis.extra_attrs();
                for attr in attrs {
                        let class = attr.get_attr_class();
                        let attr_type = class.get_type();
                        match attr_type {
                                AttrType::Foreground => {
                                        let color_attr = pango::AttrColor::from(attr);
                                        let color = color_attr.get_color();
                                        println!("red color {:?}", color.red());
                                }
                                _ => {}
                        }
                }
```
Lmk what you think

**Comment by romanzes** [comment]
Nice! Your approach is so much cleaner and simpler! I don't think my code is needed here, so I'll rebase your changes on top of master and will try to merge them to `pango-rs` on your behalf.

**Comment by romanzes** [comment]
created a PR in the main repo: https://github.com/gtk-rs/pango/pull/168

**Comment by velyan** [comment]
Awesome! 🎉 
I normally merge into master of our fork and then create a PR. Our project uses the fork anyway. 
So feel free to merge and then just keep in sync with the upstream repo. The idea is to keep in syn so that when we no longer need to add stuff, we can easily switch to the original repo.

**Pull request closed without merging**